### PR TITLE
Fix imcompatiblity of transaction side effect and data race

### PR DESCRIPTION
**Pull request by leventeliu**
Database creation transaction is also added in this PR but not used.

### CODE REVIEW

File: blockproducer/interfaces/transaction.go [hunk 1]
```diff
@@ -59,8 +59,10 @@ const (
 	TransactionTypeAlterDatabaseUser
 	// TransactionTypeDeleteDatabaseUser defines database user deletion transaction type.
 	TransactionTypeDeleteDatabaseUser
-	// TransactionTypeBaseAccount defines base account.
+	// TransactionTypeBaseAccount defines base account transaction type.
 	TransactionTypeBaseAccount
+	// TransactionTypeDeleteDatabaseUser defines database creation transaction type.
```
Comment (covenantdb) [inline blockproducer/interfaces/transaction.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported const TransactionTypeCreataDatabase should be of the form "TransactionTypeCreataDatabase ..."

File: blockproducer/runtime.go [hunk 2]
```diff
@@ -155,6 +157,18 @@ func (r *rt) getPeers() *kayak.Peers {
 	return &peers
 }
 
+func (r *rt) getHead() *State {
```
Comment (auxten) [inline blockproducer/runtime.go:160]: Maybe here needs a copy of head.

Comment (leventeliu) [reply blockproducer/runtime.go:160]: It's implied that the returned state should be readonly here, and also the State struct has its own mutex.

**Comment by xq262144** [comment]
Duplicated #89

**Pull request closed without merging**